actors/user/src/main/java/org/sunbird/user/actors/CertificateActor.java

```diff
@@ -317,32 +324,56 @@ private Map getCertificateDetails(String certificatedId) {
   }
 
   private void addCertificate(Request request) throws JsonProcessingException {
+    Response response = null;
     Map<String, String> storeMap = new HashMap<>();
+    Map<String, Object> telemetryMap = new HashMap();
     Map<String, Object> certAddReqMap = request.getRequest();
     String userId = (String) certAddReqMap.get(JsonKey.USER_ID);
+    String oldCertId = (String)certAddReqMap.get(JsonKey.OLD_ID);
     User user = userService.getUserById(userId);
     assureUniqueCertId((String) certAddReqMap.get(JsonKey.ID));
+    if(null != oldCertId) {
+      getCertificateDetails(oldCertId);
+    }
     populateStoreData(storeMap, certAddReqMap);
     certAddReqMap.put(JsonKey.STORE, storeMap);
     certAddReqMap = getRequiredRequest(certAddReqMap);
     certAddReqMap.put(JsonKey.CREATED_AT, getTimeStamp());
-    Response response =
-        cassandraOperation.insertRecord(
-            certDbInfo.getKeySpace(), certDbInfo.getTableName(), certAddReqMap);
+    if(StringUtils.isNotBlank(oldCertId)) {
```

Comment:
above we already have the check , so we can put this code there itself.

Reply:
yes no need second condition again.